genesyscloud/util_basesetting_properties.go

```diff
@@ -24,6 +24,33 @@ func buildBaseSettingsProperties(d *schema.ResourceData) *map[string]interface{}
 	return &returnValue
 }
 
+func setNullableValue[T any](d *schema.ResourceData, key string, value *T) {
```

Comment:
Nice to see some generics in this project!
These could go in a new file though. This file was just created to add some helpers for a particular resource.